huaweicloud/services/acceptance/eg/data_source_huaweicloud_eg_traced_events_test.go

```diff
@@ -0,0 +1,215 @@
+		dcBySourceName = acceptance.InitDataSourceCheck(bySourceName)
+
+		byEventType   = "data.huaweicloud_eg_traced_events.filter_by_event_type"
+		dcByEventType = acceptance.InitDataSourceCheck(byEventType)
+
+		bySubscriptionName   = "data.huaweicloud_eg_traced_events.filter_by_subscription_name"
+		dcBySubscriptionName = acceptance.InitDataSourceCheck(bySubscriptionName)
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckProjectID(t)
+			acceptance.TestAccPreCheckFgFunctionName(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceTracedEvents_basic(name, startTime.Format(time.RFC3339), endTime.Format(time.RFC3339), eventType),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(all, "region"),
+					resource.TestCheckResourceAttrSet(all, "channel_id"),
+					resource.TestCheckResourceAttrSet(all, "start_time"),
+					resource.TestCheckResourceAttrSet(all, "end_time"),
+					resource.TestMatchResourceAttr(all, "events.#", regexp.MustCompile(`^[0-9]*$`)),
+
+					dcByEventId.CheckResourceExists(),
+					resource.TestCheckOutput("is_event_id_filter_useful", "true"),
+
+					dcBySourceName.CheckResourceExists(),
+					resource.TestCheckOutput("is_source_name_filter_useful", "true"),
+
+					dcByEventType.CheckResourceExists(),
+					resource.TestCheckOutput("is_event_type_filter_useful", "true"),
+
+					dcBySubscriptionName.CheckResourceExists(),
+					resource.TestCheckOutput("is_subscription_name_filter_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceTracedEvents_base(name, eventType string) string {
+	targetID, _ := uuid.GenerateUUID()
+
+	return fmt.Sprintf(`
+resource "huaweicloud_eg_custom_event_channel" "test" {
+  name = "%[1]s"
+}
+
+resource "huaweicloud_eg_custom_event_source" "test" {
+  channel_id = huaweicloud_eg_custom_event_channel.test.id
+  name       = "%[1]s"
+}
+
+resource "huaweicloud_eg_event_subscription" "test" {
+  depends_on = [
+    huaweicloud_eg_custom_event_source.test,
+    huaweicloud_eg_custom_event_channel.test,
+  ]
+
+  channel_id  = huaweicloud_eg_custom_event_channel.test.id
+  name        = "%[1]s"
+  description = "Created by acceptance test"
+
+  sources {
+    provider_type = "CUSTOM"
+    id            = huaweicloud_eg_custom_event_source.test.id
+    name          = huaweicloud_eg_custom_event_source.test.name
+    detail        = jsonencode({})
+    filter_rule   = jsonencode({
+      "source" : [{
+        "op" : "StringIn",
+        "values" : ["${huaweicloud_eg_custom_event_source.test.name}"]
+      }]
+    })
+  }
+
+  targets {
+    id            = "%[2]s"
+    name          = "HC.FunctionGraph"
+    provider_type = "OFFICIAL"
+    detail_name   = "detail"
+    detail        = jsonencode({
+      "urn":"urn:fss:%[3]s:%[4]s:function:default:%[5]s:latest",
```

Comment:
```suggestion
      "urn": huaweicloud_fgs_function.test.urn,
```
create a new funtion in your base codes and refer in here.

Reply:
done